app/services/notification_service/course_subjects_updated.rb

```diff
@@ -0,0 +1,44 @@
+module NotificationService
+  class CourseSubjectsUpdated
+    include ServicePattern
+
+    def initialize(
+      course:,
+      previous_subject_names:,
+      previous_course_name:
+    )
+      @course = course
+      @previous_subject_names = previous_subject_names
+      @previous_course_name = previous_course_name
+    end
+
+    def call
+      return false unless notify_accredited_body?
+      return false unless course.in_current_cycle?
+
+      users.each do |user|
+        CourseSubjectsUpdatedEmailMailer.course_subjects_updated_email(
+          course: course,
+          previous_subject_names: previous_subject_names,
+          previous_course_name: previous_course_name,
+          recipient: user,
+        ).deliver_later(queue: "mailer")
+      end
+    end
+
+  private
+
+    attr_reader :course, :previous_subject_names, :previous_course_name
+
+    def users
+      @users ||= User.joins(:user_notifications).merge(UserNotification.course_update_notification_requests(course.accredited_body_code))
```

Comment:
Would it make sense to move this into `User`? Normally it makes sense for a service to coordinate between multiple models, but in this case the `UserNotification` is kind of an extension to `User`.

Reply:
Good idea @misaka 
Done -> https://github.com/DFE-Digital/teacher-training-api/pull/1478/commits/3b48730bd9a2f20d10ae0f1d5374ac8016d1a2ed